### PR TITLE
Add Mainnet Fork Tests

### PR DESCRIPTION
**Pull request by CheyenneAtapour**
Adds mainnet fork tests to Steward CCIP Extension

### CODE REVIEW

File: src/test/TestGhoStewards.t.sol [hunk 1]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+    changePrank(OWNER);
+    GHO_BUCKET_STEWARD.setControlledFacilitator(controlledFacilitators, true);
 
-    // TODO: Find which contracts are already deployed that we need
-    // TODO: Deploy stewards, using corresponding contracts
-    // TODO: Ensure we grant all appropriate roles
+    vm.stopPrank();
   }
 
   function testSetup() public {
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:106]:
```suggestion
  function testStewardsPermissions() public {
```

File: src/test/TestGhoStewards.t.sol
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:1]: Rename to something along the lines of `TestGhoStewardsForked`

File: src/test/TestGhoStewards.t.sol [hunk 2]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+    GHO_AAVE_STEWARD.updateGhoBorrowCap(newBorrowCap);
+    assertEq(_getGhoBorrowCap(), newBorrowCap);
+  }
+
+  function testGhoAaveStewardUpdateGhoSupplyCap() public {
+    uint256 currentSupplyCap = _getGhoSupplyCap();
+    assertEq(currentSupplyCap, 0);
+    uint256 newSupplyCap = currentSupplyCap + 1;
+    // Can't update supply cap even by 1 since it's 0, and 100% of 0 is 0
+    vm.expectRevert('INVALID_SUPPLY_CAP_UPDATE');
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:155]: revisit spec

File: src/test/TestGhoStewards.t.sol [hunk 3]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+  function testGhoAaveStewardUpdateGhoBorrowRate() public {
+    IDefaultInterestRateStrategyV2.InterestRateData memory currentRates = _getGhoBorrowRates();
+    vm.prank(RISK_COUNCIL);
+    GHO_AAVE_STEWARD.updateGhoBorrowRate(
+      currentRates.optimalUsageRatio,
+      currentRates.baseVariableBorrowRate + 1,
+      currentRates.variableRateSlope1,
+      currentRates.variableRateSlope2
+    );
+    assertEq(_getGhoBorrowRate(), currentRates.baseVariableBorrowRate + 1);
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:169]: check other params as well please. Can be done in the same test

File: src/test/TestGhoStewards.t.sol [hunk 4]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+  function testGhoBucketStewardSetControlledFacilitatorAdd() public {
+    address[] memory oldControlledFacilitators = GHO_BUCKET_STEWARD.getControlledFacilitators();
+    address[] memory newGsmList = new address[](1);
+    address gho_gsm_4626 = makeAddr('gho_gsm_4626');
+    newGsmList[0] = gho_gsm_4626;
+    vm.prank(OWNER);
+    GHO_BUCKET_STEWARD.setControlledFacilitator(newGsmList, true);
+    address[] memory newControlledFacilitators = GHO_BUCKET_STEWARD.getControlledFacilitators();
+    assertEq(newControlledFacilitators.length, oldControlledFacilitators.length + 1);
+    assertTrue(_contains(newControlledFacilitators, gho_gsm_4626));
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:192]: this indicates we need to add an `isControlledFacilitator` getter

Comment (miguelmtzinf) [reply src/test/TestGhoStewards.t.sol:192]: try to remove as well pls, in the same test

File: src/test/TestGhoStewards.t.sol [hunk 5]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+      address(GHO_ATOKEN)
+    );
+    vm.prank(RISK_COUNCIL);
+    uint128 newBucketCapacity = uint128(currentBucketCapacity) + 1;
+    GHO_BUCKET_STEWARD.updateFacilitatorBucketCapacity(address(GHO_ATOKEN), newBucketCapacity);
+    (uint256 capacity, ) = GhoToken(GHO_TOKEN).getFacilitatorBucket(address(GHO_ATOKEN));
+    assertEq(newBucketCapacity, capacity);
+  }
+
+  function testGhoBucketStewardSetControlledFacilitatorAdd() public {
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:183]:
```suggestion
  function testGhoBucketStewardSetControlledFacilitator() public {
```

File: src/test/TestGhoStewards.t.sol [hunk 6]
```diff
@@ -40,45 +65,231 @@ contract TestGhoStewards is Test {
...
+    });
+
+    RateLimiter.Config memory newInboundConfig = RateLimiter.Config({
+      isEnabled: outboundConfig.isEnabled,
+      capacity: inboundConfig.capacity,
+      rate: inboundConfig.rate
+    });
+
+    // Currently rate limit set to 0, so can't even change by 1 because 100% of 0 is 0
+    vm.expectRevert('INVALID_RATE_LIMIT_UPDATE');
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewards.t.sol:225]: revisit spec

File: src/test/TestGhoStewards.t.sol (hunk 1 above)
Comment (CheyenneAtapour) [reply src/test/TestGhoStewards.t.sol]: Done

File: src/test/TestGhoStewards.t.sol
Comment (CheyenneAtapour) [reply src/test/TestGhoStewards.t.sol:1]: Done

File: src/test/TestGhoStewards.t.sol (hunk 3 above)
Comment (CheyenneAtapour) [reply src/test/TestGhoStewards.t.sol]: Done

File: src/test/TestGhoStewards.t.sol (hunk 4 above)
Comment (CheyenneAtapour) [reply src/test/TestGhoStewards.t.sol]: Done

File: src/test/TestGhoStewards.t.sol (hunk 5 above)
Comment (CheyenneAtapour) [reply src/test/TestGhoStewards.t.sol]: Done

File: src/test/TestGhoStewardsForkEthMainnet.t.sol
Comment (miguelmtzinf) [inline src/test/TestGhoStewardsForkEthMainnet.t.sol:1]: I d rename to "TestGhoStewardsForkEthereum.t.sol"
and create another one "TestGhoStewardsForkRemote", which can use the Arbitrum config
so we cover both sides

Comment (CheyenneAtapour) [reply src/test/TestGhoStewardsForkEthMainnet.t.sol:1]: Done

File: src/test/TestGhoStewardsForkRemote.t.sol [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+    GHO_BUCKET_STEWARD = new GhoBucketSteward(OWNER, GHO_TOKEN, RISK_COUNCIL);
+    GhoToken(GHO_TOKEN).grantRole(
+      GhoToken(GHO_TOKEN).BUCKET_MANAGER_ROLE(),
+      address(GHO_BUCKET_STEWARD)
+    );
+
+    GHO_CCIP_STEWARD = new GhoCcipSteward(GHO_TOKEN, GHO_TOKEN_POOL, RISK_COUNCIL, true);
+
+    address[] memory controlledFacilitators = new address[](1);
+    controlledFacilitators[0] = address(GHO_ATOKEN);
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewardsForkRemote.t.sol:80]: GHO ATOKEN is not a facilitator in remote deployments

File: src/test/TestGhoStewardsForkRemote.t.sol [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+    GHO_CCIP_STEWARD.updateRateLimit(
+      remoteChainSelector,
+      newOutboundConfig.isEnabled,
+      newOutboundConfig.capacity,
+      newOutboundConfig.rate,
+      newInboundConfig.isEnabled,
+      newInboundConfig.capacity,
+      newInboundConfig.rate
+    );
+  }
```
Comment (miguelmtzinf) [inline src/test/TestGhoStewardsForkRemote.t.sol:298]: assert new rate config is applied

Comment (CheyenneAtapour) [reply src/test/TestGhoStewardsForkRemote.t.sol:298]: I can do that, but I have the event emission checking easily in one line, which is the pattern I'd been using for these tests

Comment (miguelmtzinf) [reply src/test/TestGhoStewardsForkRemote.t.sol:298]: ah, it's fine then

File: src/test/TestGhoStewardsForkRemote.t.sol (hunk 7 above)
Comment (CheyenneAtapour) [reply src/test/TestGhoStewardsForkRemote.t.sol]: Done

File: src/test/TestGhoStewards.t.sol (hunk 6 above)
Comment (miguelmtzinf) [reply src/test/TestGhoStewards.t.sol]: This is correct, going from zero to non-zero is something that requires AIP process.

**Pull request merged**